src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -178,21 +178,24 @@ get_device_id(const std::string& bdf) const
   if (bdf.find_first_not_of("0123456789") == std::string::npos)
     return system::get_device_id(bdf);
 
-  unsigned int i = 0;
-  for (auto dev = get_pcidev(0); dev; dev = get_pcidev(++i)) {
+  try {
+    for (unsigned int i = 0;; i++) {
+      auto dev = get_pcidev(i);
```

Comment:
The reason I asked about throw or nullptr is this code.  It is not cool to rely on exceptions in success path, which is  kind of what this code is doing in order to break the loop.    It's nitpicking, but still.

Reply:
I agree soren but the reason why I have implemented this way is because there are lot of functions which directly use pointer returned from get_pcidev() function without checking it for non null. In order to eliminate lot of changes I used this way.

Reply:
I see and in these cases the emitted exception is eventually caught?

Reply:
Yes Soren, upper layer catches the exception thrown